plugins/tabular_dataset.cc

```diff
@@ -607,25 +1230,70 @@ commit()
     return itl->commit();
 }
 
+Dataset::MultiChunkRecorder
+TabularDataset::
+getChunkRecorder()
+{
+    MultiChunkRecorder result;
+    result.newChunk = [=] (size_t)
+        {
+            return std::unique_ptr<Recorder>
+                (new TabularDataStore::ChunkRecorder(itl.get()));
+        };
+
+    result.commit = [=] () { this->commit(); };
+    return result;
+}
+
 void
 TabularDataset::
 recordRowItl(const RowName & rowName,
              const std::vector<std::tuple<ColumnName, CellValue, Date> > & vals)
 {
     validateNames(rowName, vals);
-    return itl->recordRow(rowName, vals);
+    itl->recordRow(rowName, vals);
 }
 
-struct TabularDatasetConfig {
+void
+TabularDataset::
+recordRows(const std::vector<std::pair<RowName, std::vector<std::tuple<ColumnName, CellValue, Date> > > > & rows)
+{
+    for (auto & r: rows)
+        itl->recordRow(r.first, r.second);
+}
+
+/*****************************************************************************/
+/* TABULAR DATASET                                                           */
+/*****************************************************************************/
+
+TabularDatasetConfig::
+TabularDatasetConfig()
+{
+    unknownColumns = UC_ERROR;
+}
+
+DEFINE_ENUM_DESCRIPTION(UnknownColumnAction);
+
+UnknownColumnActionDescription::
+UnknownColumnActionDescription()
+{
+    addValue("ignore", UC_IGNORE, "Unknown columns will be ignored");
+    addValue("error", UC_ERROR, "Unknown columns will result in an error");
+    addValue("add", UC_ADD, "Unknown columns will be added as a sparse column");
 };
 
-DECLARE_STRUCTURE_DESCRIPTION(TabularDatasetConfig);
 DEFINE_STRUCTURE_DESCRIPTION(TabularDatasetConfig);
 
 TabularDatasetConfigDescription::
 TabularDatasetConfigDescription()
 {
     nullAccepted = true;
+
```

Comment:
Yes, the missing columns will have nulls inferred.

The issue with that parameter is that it will be confusing for a user with respect to the importer function and the select expression.  We're giving them 3 different places to write column names, without any clear way to know where to put them.